### PR TITLE
activeClassName as component for link.

### PR DESCRIPTION
**Pull request by hiteshjoshi**
This lets you add,` <Link activeClassName="is-active" href="/about" />`
Small hack, but works well.

### CODE REVIEW

**Comment by developit** [comment]
@hiteshjoshi love this idea, but I think we'll need to make `Link` a `class` component - right now something has to re-render it in order to have it re-check the URL.  A component outside of the router (like a Header) wouldn't be updated when the router's URL changes.  Need to think about how that would work :)

**Comment by hiteshjoshi** [comment]
That's what I thought, but silly it updates even when outer component is changed.

**Comment by developit** [comment]
Yup, that's why it was updating.  Definitely not a bad practise to keep router at the root, it just depends how you want to set your app up.  There are great reasons to keep it that high up - if you want your page layout to be able to change entirely between routes, it's definitely the best way to go.

**Comment by tauren** [comment]
What needs to be done to get this completed? Can I help?

I'm trying to move a simple app from react + react-router to preact + preact-router. For the most part, I have everything working without a lot of effort. But something as simple as styling the "active" option in the main menu bar is proving to require a lot of extra work. I'd love to see preact-router support this natively.

File: src/index.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ function setUrl(url, type='push') {
 }
 
 
-function getCurrentUrl() {
+function getCurrentUrl(search=true) {
 	let url;
```
Comment (lukeed) [inline src/index.js]:
```js
let url, search = search || false;
```

**Pull request merged**